libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/WebViewStateManager.swift

```diff
@@ -0,0 +1,102 @@
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+//  DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+//  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+//  WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+//  WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import Foundation
+import WebKit
+
+@objc(SFSDKWebViewStateManager)
+public class WebViewStateManager: NSObject {
+    private static var processPool: WKProcessPool?
+    private static var sessionCookieManagementDisabled = false
+
+    @objc
+    @MainActor
+    public static func removeSession() {
+        sharedProcessPool = nil
+
+        if sessionCookieManagementDisabled {
+            SFSDKCoreLogger.d(WebViewStateManager.self, message: "[\(Self.self) removeSession]: Cookie Management disabled. Will do nothing.")
+            return
+        }
+        
+        // Perform async cleanup without blocking the current thread
+        Task {
+            await removeWKWebViewCookies()
+        }
+    }
+
+    @objc
+    @MainActor
+    public static func removeSessionForcefully() async {
+        sharedProcessPool = nil
+        await removeWKWebViewCookies()
+    }
+
+    @objc
+    public static func resetSessionCookie() async {
+        if sessionCookieManagementDisabled {
+            SFSDKCoreLogger.d(WebViewStateManager.self, message: "[\(Self.self) resetSessionCookie]: Cookie Management disabled. Will do nothing.")
+            return
+        }
+        await removeWKWebViewCookies()
+    }
+
+    @MainActor
+    private static func removeWKWebViewCookies() async {
+        await withCheckedContinuation { continuation in
+            let dataStore = WKWebsiteDataStore.default()
+            let websiteDataTypes: Set<String> = [WKWebsiteDataTypeCookies]
+            dataStore.removeData(ofTypes: websiteDataTypes, modifiedSince: Date.distantPast) {
+                continuation.resume()
+            }
+        }
+        
+    }
+
+    @objc
+    @MainActor
+    public static var sharedProcessPool: WKProcessPool? {
+        get {
+            if processPool == nil {
+                SFSDKCoreLogger.i(WebViewStateManager.self, message: "[\(Self.self) sharedProcessPool]: No process pool exists. Creating new instance.")
+                processPool = WKProcessPool()
+            }
+            return processPool
+        }
+        set {
+            if newValue !== processPool {
+                SFSDKCoreLogger.i(WebViewStateManager.self, message: "[\(Self.self) setSharedProcessPool]: Changing from \(String(describing: processPool)) to \(String(describing: newValue))")
+                processPool = newValue
+            }
+        }
+    }
+
+    @objc
+    public static func setSessionCookieManagementDisabled(_ disabled: Bool) {
```

Comment:
It looks like this and the method below are a breaking API change too since it was a property before. Could it be a `static var` or was there an issue with it?

Reply:
Ah, yes.  I'll fix that too.  Thank you.